### DISCUSSION TITLE
Equal axes

### DISCUSSION THREAD

**Discussion by brenocq** [brenocq/implot3d#22] [General]
### This discussion was converted into issue #104

Right now the zoom of each axis is independent, but we should be able to force the axes to zoom together. This task would introduce the `ImPlot3D_Equal` flag to the plot flags.

**Comment by dimateos** [comment]
Probably what may confuse most users is the linked zoom for 2 simultaneous dimensions when the mouse is hovering a plane.
I like being able to zoom per axis, but maybe I prefer to zoom over all when hovering no axis. So I guess I would add also the flag `ImPlot3D_Disable2DZoom` (with a better name picked by u xD) 

Btw there is a lot of potential for modifier keys as suggested (kind of) here https://github.com/brenocq/implot3d/discussions/38
So for example the zoom could be global by default, and per axis/plane only when the user is holding CTRL or SHIFT

**Reply by brenocq** [reply]
> So I guess I would add also the flag ImPlot3D_Disable2DZoom (with a better name picked by u xD)

I like the idea! Having no 2D zoom is indeed the default for many other 3D visualization libraries. I'll add this flag this week :)

> So for example the zoom could be global by default, and per axis/plane only when the user is holding CTRL or SHIFT

Cool idea! I would love to get the input from more users on this :)

**Comment by brenocq** [comment]
Hi @dimateos, I'm converting all feature discussions to issues since GitHub has better support for task tracking (and copilot agent execution) with issues. This discussion was converted into #104
